Update observability_enable.adoc

dockerymick opened the pull request:
updates
https://github.com/open-cluster-management/backlog/issues/12628#issuecomment-854245667

dockerymick committed:
Update observability_enable.adoc

updates
https://github.com/open-cluster-management/backlog/issues/12628#issuecomment-854245667

marcolan018 commented on observability/observability_enable.adoc:
```diff
@@ -176,58 +176,15 @@ View the following default YAML file for observability:
-    storeStorageSize: 10Gi # The amount of storage applied to thanos store stateful sets
-  status: #Status contains the different condition statuses for this deployment
+      key: thanos.yaml
+  status: {}
```
needs to remove "status: {}"
this one is not set by user, but set by our component

dockerymick committed:
Update observability_enable.adoc

update after Marco's review

marcolan018 commented on observability/observability_enable.adoc:
```diff
@@ -176,58 +176,14 @@ View the following default YAML file for observability:
-    metricObjectStorage: # Configuring access to object storage
-      key: thanos.yaml
+  observabilityAddonSpec: {}
+  advanced: {}
```
don't need to add the empty advanced

dockerymick committed:
Update observability_enable.adoc

removed advanced parameter

dockerymick committed:
Update whats_new.adoc

updating the reference link and other parts of what's new so that there is no merge conflict

dockerymick committed:
Merge branch '2.3_stage' into dockerymick-patch-1
